storage/reads/stream_reader.go

```diff
@@ -340,6 +349,9 @@ func (r *frameReader) init() {
 }
 
 func (r *frameReader) peekFrame() *datatypes.ReadResponse_Frame {
+	retries := peekFrameRetries
+
+RETRY:
```

Comment:
If we have a previous `r.buf` and retry, won't `r.p == 0` so we just return the previous buffer?

Reply:
You are right – I'll clear r.buf on retry. Great catch!

Reply:
I created a test to catch that, however, I was reminded `Recv` always returns a non-nil value, so r.buf was always being overwritten. Regardless, I am explicitly clearing for safety.